### PR TITLE
Add debug print in loss functions for link prediction debugging

### PR DESCRIPTION
**Pull request by classicsong**
*Issue #, if available:*
The adversarial negative loss function for link prediction and contrastive loss function for link prediction are sensitive to hyperparameters.

*Description of changes:*
Add some debug print to help people better debug abnormal link prediction scores and losses, such as inf or NaN.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/graphstorm/model/loss_func.py [hunk 1]
```diff
@@ -294,17 +297,37 @@ def __init__(self, adversarial_temperature):
...
-                                 dim=-1).detach() * loss, dim=-1)
+        loss = th.softmax(score * self.adversarial_temperature,
+                                 dim=-1).detach() * loss
+        # For debugging abnormal loss values.
+        if self._debug_print and get_rank() == 0:
+            logging.debug("Adversarial cross entropy loss for negative samples \
+                          with input score as {} and loss as {}. \
+                          For abnormal loss values, it is suggested to enable \
+                          lp embed normalizer by setting --lp-embed-normalizer l2_norm \
+                          and tune the hyperparameter gamma.".format(
```
Comment (Oxfordblue7) [inline python/graphstorm/model/loss_func.py:325]:
```suggestion
                          and tune the hyperparameter adversarial_temperature.".format(
```
Tuning gamma does not work.

File: python/graphstorm/model/loss_func.py [hunk 2]
```diff
@@ -447,8 +471,19 @@ def forward(self, pos_score, neg_score):
...
         exp_logits = th.exp(score)
         log_prob = pscore - th.log(exp_logits.sum(1))
+        loss = -log_prob.mean()
+
+        # For debugging abnormal loss values.
+        if self._debug_print and get_rank() == 0:
+            logging.debug("Contrastive loss with pos score as {} \
+                          and neg score as {} and loss as {}.\
+                          For abnormal loss values, it is suggested \
+                          to tune the hyperparameter gamma.or \
```
Comment (Oxfordblue7) [inline python/graphstorm/model/loss_func.py:481]:
```suggestion
                          to tune the hyperparameter \
```

File: python/graphstorm/model/loss_func.py [hunk 3]
```diff
@@ -447,8 +471,19 @@ def forward(self, pos_score, neg_score):
...
+                          the contrastive_loss_temperature.".format(
```
Comment (Oxfordblue7) [inline python/graphstorm/model/loss_func.py:482]:
```suggestion
                         contrastive_loss_temperature.".format(
```

File: tests/end2end-tests/graphstorm-lp/test.sh [hunk 4]
```diff
@@ -158,7 +158,7 @@ error_and_exit $?
...
-python3 -m graphstorm.run.gs_link_prediction --workspace $GS_HOME/training_scripts/gsgnn_lp --num-trainers $NUM_TRAINERS --num-servers 1 --num-samplers 0 --part-config /data/movielen_100k_lp_train_val_1p_4t/movie-lens-100k.json --ip-config ip_list.txt --ssh-port 2222 --cf ml_lp.yaml --fanout '10,15' --num-layers 2 --use-mini-batch-infer false --lp-decoder-type rotate --train-etype user,rating,movie movie,rating-rev,user --num-epochs 1 --eval-frequency 300 --contrastive-loss-temperature 0.01 --lp-loss-func contrastive
+python3 -m graphstorm.run.gs_link_prediction --workspace $GS_HOME/training_scripts/gsgnn_lp --num-trainers $NUM_TRAINERS --num-servers 1 --num-samplers 0 --part-config /data/movielen_100k_lp_train_val_1p_4t/movie-lens-100k.json --ip-config ip_list.txt --ssh-port 2222 --cf ml_lp.yaml --fanout '10,15' --num-layers 2 --use-mini-batch-infer false --lp-decoder-type rotate --train-etype user,rating,movie movie,rating-rev,user --num-epochs 1 --eval-frequency 300 --contrastive-loss-temperature 0.01 --lp-loss-func contrastive --contrastive-loss-temperature 0.1 --logging-level debug
```
Comment (Oxfordblue7) [inline tests/end2end-tests/graphstorm-lp/test.sh:161]:
```suggestion
python3 -m graphstorm.run.gs_link_prediction --workspace $GS_HOME/training_scripts/gsgnn_lp --num-trainers $NUM_TRAINERS --num-servers 1 --num-samplers 0 --part-config /data/movielen_100k_lp_train_val_1p_4t/movie-lens-100k.json --ip-config ip_list.txt --ssh-port 2222 --cf ml_lp.yaml --fanout '10,15' --num-layers 2 --use-mini-batch-infer false --lp-decoder-type rotate --train-etype user,rating,movie movie,rating-rev,user --num-epochs 1 --eval-frequency 300 --contrastive-loss-temperature 0.1 --lp-loss-func contrastive --logging-level debug
```

File: python/graphstorm/model/loss_func.py
Comment (Oxfordblue7) [inline python/graphstorm/model/loss_func.py:1]: Do we need to add docs for this debugging feature?

File: python/graphstorm/model/loss_func.py (hunk 1 above)
Comment (classicsong) [reply python/graphstorm/model/loss_func.py:325]: tunning gamma also works.

File: python/graphstorm/model/loss_func.py
Comment (classicsong) [reply python/graphstorm/model/loss_func.py:1]: Not now.

**Review by Oxfordblue7 — APPROVED** [review]
